### PR TITLE
Add CommentToPhpdocFixer

### PR DESCRIPTION
**Pull request by kubawerlos**
As mentioned by @Slamdunk in https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3439#issuecomment-357189286 here is a proposition of fixer
```diff
-/*
+/**
 * @var string $foo
 */
```
when the comment contains annotation.

### CODE REVIEW

**Comment by keradus** [comment]
please add to `.php_cs.dist`

**Comment by kubawerlos** [comment]
done

File: tests/Fixer/Comment/CommentToPhpdocFixerTest.php [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+    public function provideTestCases()
+    {
+        return [
+            [
+                '<?php /* string $foo */',
+            ],
+            [
+                '<?php /* $yoda string @var */',
```
Comment (keradus) [inline tests/Fixer/Comment/CommentToPhpdocFixerTest.php]: please add

```
+            [
+                '<?php /* $yoda @var string */',
+            ],
```

File: tests/Fixtures/Integration/priority/comment_to_phpdoc,phpdoc_to_comment.test [hunk 2]
```diff
@@ -0,0 +1,12 @@
+--TEST--
+Integration of fixers: comment_to_phpdoc,phpdoc_to_comment.
+--RULESET--
+{"comment_to_phpdoc": true, "phpdoc_to_comment": true}
```
Comment (keradus) [inline tests/Fixtures/Integration/priority/comment_to_phpdoc,phpdoc_to_comment.test]: How do you show that order of fixers matters (you put this in priority dir), if there is no input code, means that none of the fixer does any job ?

Comment (kubawerlos) [reply tests/Fixtures/Integration/priority/comment_to_phpdoc,phpdoc_to_comment.test]: I've added other comment which would change.

**Comment by kubawerlos** [comment]
@keradus test cases added.

I've added `CommentsAnalyzer` I indent to use in `NoEmptyCommentFixer` after https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3455 is merged to `master`.

File: src/Console/WarningsDetector.php [hunk 3]
```diff
@@ -39,8 +39,10 @@ public function __construct(ToolInfoInterface $toolInfo)
 
     public function detectOldMajor()
     {
-        // @TODO 3.0 to be activated with new MAJOR release
-        // $this->warnings[] = 'You are running PHP CS Fixer v2, which is not maintained anymore. Please update to v3.';
+        /*
```
[the remark this answers is not in the document]
Comment (kubawerlos) [reply src/Console/WarningsDetector.php]: How? This fixer is in `.php_cs.dist` and is "merging" consecutive single line comments into one -  I cannot see a way how we can exclude this scenarion - other than mixing this fixer with `PhpdocToCommentFixer`.

File: tests/Fixer/Comment/CommentToPhpdocFixerTest.php [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+# This is my var
+# @var string $foo
+# stop using it
+# @deprecated since 1.3
+$foo = 1;
+EOT
+                ,
+            ],
+            [
```
[the remark this answers is not in the document]
Comment (kubawerlos) [reply tests/Fixer/Comment/CommentToPhpdocFixerTest.php]: Indeed it should.

File: src/Console/WarningsDetector.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (julienfalque) [reply src/Console/WarningsDetector.php]: What I meant is that the fixer is supposed to transform regular comments into PHPDoc, but I don't think it's relevant here: the subsequent tokens are not supposed to have a PHPDoc. Also, the new comment is a multiline comment but *not* a DocComment.

[the remark this answers is not in the document]
Comment (keradus) [reply src/Console/WarningsDetector.php]: I don't believe that fixer could distinguish this edge case, when we have consecutive comment that looks like docblock and commented code.
For that, one has to manually adjust the code to looks fine.

Also, concern about comment not being docblock in the end is still valid.

**Comment by keradus** [comment]
Thank you @kubawerlos.

**Pull request closed without merging**